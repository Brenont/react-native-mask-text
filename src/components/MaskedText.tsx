import React from "react";
import { Text, TextProps } from "react-native";
import { mask } from "../utils/mask";

interface MaskedTextProps {
  children: string;
  mask: string;
}

export default function MaskedText({
  children: text,
  mask: pattern,
  ...rest
}: MaskedTextProps & TextProps) {
  return <Text {...rest}>{mask(text, pattern)}</Text>;
}
