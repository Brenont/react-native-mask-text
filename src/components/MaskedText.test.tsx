import { render } from '@testing-library/react-native';
import { MaskedText } from './MaskedText';

describe('<MaskedText />', () => {
  test('should renders correctly', () => {
    const container = render(<MaskedText mask="AAA-999">RCT777</MaskedText>);

    expect(container.getByText('RCT-777')).toBeTruthy();

    expect(container).toMatchSnapshot();
  });
})