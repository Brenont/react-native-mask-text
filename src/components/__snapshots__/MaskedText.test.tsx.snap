// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MaskedText /> should renders correctly 1`] = `
<Text>
  RCT-777
</Text>
`;
